packages/ui/src/components/countDown/CountDown.component.js

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import * as R from 'ramda';
+
+import * as Styled from './countDown.styled';
+
+const defaultFormatter = (time) => {
+  return Math.round(time / 1000);
+};
+
+const CountDown = ({ total, counterRef, onFinished, formatter = defaultFormatter, interval = 1000, children }) => {
+  const [time, setTime] = useState();
+  const [resetVersion, setResetVersion] = useState(0);
+
+  if (counterRef && 'current' in counterRef) {
+    counterRef.current = {
+      reset: () => {
+        setResetVersion(resetVersion + 1);
+      },
+    };
+  }
+
+  useEffect(() => {
+    setTime(total);
+    const handle = setInterval(() => {
+      setTime((time) => {
+        if (time >= interval) {
+          return time - interval;
+        } else {
+          clearInterval(handle);
+          R.is(Function, onFinished) && onFinished();
```

Comment:
The type check is probably redundant as the onFinished has been declared as `PropTypes.func`.